src/intro.tex

```diff
@@ -165,6 +165,15 @@ \section{RISC-V Software Execution Environments and Harts}
 Some EEIs support the creation and destruction of additional harts,
 for example, via environment calls to fork new harts.
 
+The execution environment is responsible for ensuring eventual
+forward progress of each of its harts.  Forward progress includes:
+\begin{itemize}
+\item Completing the execution of an instruction.
+\item Other events defined by extensions, such as traps.
+\item Mechanisms that explicitly wait for an event, such as the WFI
+  instruction waiting for interrupts, or various environment calls.
+\end{itemize}
```

Comment:
I think reference to or accommodation of the trap taxonomy in Section 1.6 is useful here.

Fatal traps and some requested traps halt the hart.

Reply:
What about rewording, "forward progress of each of its harts" to "forward progress of each of its harts until they terminate" ?

Reply:
I think we should leverage existing formal terms.  Terminate isn't formally defined, but is used in the description of "Fatal Trap".

Reply:
It's a good point.  I looked and found that Section 1.6 specifically calls out that Requested Traps may request termination (footnote 1 to Table 1.1).  So neither "Fatal Trap" nor "Requested Trap" exactly fit the bill, and both use the termination terminology which seems pretty clear.  Saying "until their termination by Fatal Traps or Requested Traps" rather than simply "until their termination" seems to weigh things down without adding clarity.  ??

Reply:
(Also, a hart or EE can terminate without needing a trap.)

Reply:
Agreed @marceg